gateway/sds_gateway/users/views.py

```diff
@@ -1800,3 +1819,562 @@ def get(self, request, *args, **kwargs) -> JsonResponse:
+        elif errors and not all_files_empty:
+            logger.error(
+                "File upload errors occurred, skipping capture creation. Errors: %s",
+                errors,
+            )
+
+        # Determine status
+        status = self._determine_response_status(
+            saved_files,
+            created_captures,
+            all_files_empty,
+            has_required_fields,
+            files,
+            errors,
+        )
+
+        response_data = self._build_response_data(
+            status,
+            saved_files,
+            created_captures,
+            files,
+            errors,
+            capture_errors,
+            all_files_empty=all_files_empty,
+            has_required_fields=has_required_fields,
+        )
+        # Return 200 if we have files saved, captures created, or if all files were
+        # skipped with required fields
+        status_code = self._determine_status_code(
+            saved_files, created_captures, all_files_empty, has_required_fields
+        )
+
+        return JsonResponse(response_data, status=status_code)
+
+
+user_upload_files_view = UploadFilesView.as_view()
+
+
+@method_decorator(csrf_exempt, name="dispatch")
+class CheckFileExistsView(View):
+    """View to check if a file exists based on path, name, and checksum."""
+
+    def post(self, request, *args, **kwargs):
+        """Check if a file exists using the provided path, name, and checksum."""
+        try:
+            # Get data from request
+            data = json.loads(request.body)
+            directory = data.get("directory", "")
+            filename = data.get("filename", "")
+            checksum = data.get("checksum", "")
+
+            # Validate required fields
+            if not all([directory, filename, checksum]):
+                return JsonResponse(
+                    {
+                        "error": (
+                            "Missing required fields: directory, filename, and "
+                            "checksum are required"
+                        )
+                    },
+                    status=400,
+                )
+
+            # Prepare data for check_file_contents_exist_helper
+            check_data = {
+                "directory": directory,
+                "name": filename,
+                "sum_blake3": checksum,
+            }
+
+            # Call the helper function
+            response = check_file_contents_exist_helper(request, check_data)
+
+            # Extract the response data
+            if hasattr(response, "data"):
+                response_data = response.data
+            else:
+                response_data = str(response)
+
+            # Return the result
+            http_ok = 200
+            return JsonResponse(
+                {
+                    "exists": response.status_code == http_ok,
+                    "status_code": response.status_code,
+                    "data": response_data,
+                }
+            )
```

Comment:
This interprets a successful response as "the file exists" (`"exists": response.status_code == http_ok`), but `CheckFileContentsExistView.post` (the method requested by the helper) will also succeed if the file doesn't exist.

See `CheckFileContentsExistView.post`'s docstring to know what to look for in the response payload. Please reach out if this comment is not clear.